content/guides/angular/containerize.md

```diff
@@ -0,0 +1,497 @@
+# =========================================
+
+FROM nginxinc/nginx-unprivileged:${NGINX_VERSION} AS runner
+
+# Use a built-in non-root user for security best practices
+USER nginx
+
+# Copy custom Nginx config
+COPY --link nginx.conf /etc/nginx/nginx.conf
+
+# Application name
+ARG APP_NAME=docker-angular-sample
+
+# Copy the static build output from the build stage to Nginx's default HTML serving directory
+COPY --link --from=builder /app/dist/${APP_NAME}/browser /usr/share/nginx/html
+
+# Expose port 8080 to allow HTTP traffic
+# Note: The default NGINX container now listens on port 8080 instead of 80 
+EXPOSE 8080
+
+# Start Nginx directly with custom config
+ENTRYPOINT ["nginx", "-c", "/etc/nginx/nginx.conf"]
+CMD ["-g", "daemon off;"]
+```
+
+### Step 3: Configure the .dockerignore file
+
+The `.dockerignore` file tells Docker which files and folders to exclude when building the image.
+
+> [!NOTE]
+>This helps:
+>- Reduce image size  
+>- Speed up the build process  
+>- Prevent sensitive or unnecessary files (like `.env`, `.git`, or `node_modules`) from being added to the final image.
+>
+> To learn more, visit the [.dockerignore reference](/reference/dockerfile.md#dockerignore-file).
+
+Copy and replace the contents of your existing `.dockerignore` with the configuration below:
+
+```dockerignore
+# ================================
+# Node and build output
+# ================================
+node_modules
+dist
+out-tsc
+.angular
+.cache
+.tmp
+
+# ================================
+# Testing & Coverage
+# ================================
+coverage
+jest
+cypress
+cypress/screenshots
+cypress/videos
+reports
+playwright-report
+.vite
+.vitepress
+
+# ================================
+# Environment & log files
+# ================================
+*.env*
+!*.env.production
+*.log
+*.tsbuildinfo
+
+# ================================
+# IDE & OS-specific files
+# ================================
+.vscode
+.idea
+.DS_Store
+Thumbs.db
+*.swp
+
+# ================================
+# Version control & CI files
+# ================================
+.git
+.gitignore
+
+# ================================
+# Docker & local orchestration
+# ================================
+Dockerfile
+Dockerfile.*
+.dockerignore
+docker-compose.yml
+docker-compose*.yml
+
+# ================================
+# Miscellaneous
+# ================================
+*.bak
+*.old
+*.tmp
+```
+
+### Step 4: Create the `nginx.conf` file
+
+To serve your Angular application efficiently inside the container, you’ll configure NGINX with a custom setup. This configuration is optimized for performance, browser caching, gzip compression, and support for client-side routing.
+
+Create a file named `nginx.conf` in the root of your project directory, and add the following content:
+
+> [!NOTE]
+> To learn more about configuring NGINX, see the [official NGINX documentation](https://nginx.org/en/docs/).
+
+
+```nginx
+worker_processes auto;
+
+pid /tmp/nginx.pid;
```

Comment:
What is the reason to use `/tmp/nginx.pid` instead of a conventional `/var/run/nginx.pid`?